source/Projectile.cpp

```diff
@@ -76,8 +76,10 @@ Projectile::Projectile(const Projectile &parent, const Weapon *weapon)
 		if(!parent.weapon->Acceleration())
 		{
 			// Move in this new direction at the same velocity.
-			double parentVelocity = parent.weapon->Velocity();
-			velocity += (this->angle.Unit() - parent.angle.Unit()) * parentVelocity;
+			// To maintain the sign of the velocity, Point::Length can’t be used.
+			Point referenceVector = parent.angle.Unit();
+			double parentVelocity = referenceVector.Dot(parent.velocity);
+			velocity = this->angle.Unit() * parentVelocity;
```

Comment:
I still have some questions on the actual desired behavior (of the code in-general, not just as changed in this PR):

1) What's the end goal, i.e. what _should_ the velocity of a new submunition be, after we've finished creating it? We have a projectile or submunition that is expiring, and is producing submunition(s). At the start of this block scope, a produced submunition `this` is traveling with the same speed and same direction as the dying parent, but is pointed in some different direction. I see two broad descriptions of how we can treat this process:
    1) `this` should be "perfectly emitted," meaning the overall magnitude of the parent velocity is applied to the new submunition's direction, 100%. (This means the game will not preserve player-perceived momentum when creating submunitions. This will be particularly noticeable if multiple submunitions are created and have different directions from each other.)
   2) `this` should be "physically emitted," meaning we apply only some portion of the parent's velocity to it. One possible implementation would be to set the submunition's velocity magnitude to be the same as the parent's projected velocity along the submunition's facing angle, e.g trading some of original momentum for orthogonal momentum. In this mode (absent any random velocity present in the new submunition's `weapon`), I think the player would observe the original velocity of the parent when viewing the motion of all created submunitions as a whole -- think of a projectile that explodes like a firework.

2) Can we deduce the reasoning behind only applying this compensation when the _parent_ projectile/submunition has no acceleration? I am left to wonder if this was meant to be performed only if the _new submunition_'s weapon has no acceleration (and thus the direction in which the submunition is traveling cannot change).

Reply:
I understand what you mean:
![projectile](https://user-images.githubusercontent.com/22392249/102010146-88313800-3d7f-11eb-8cfa-997aabce6a1d.png)
i. and ii. are not too different, if "inaccuracy" is ordinarily range.

I guess it's the reason why a projectile that has acceleration does not apply this compensation that the projectile is probably a physical missile on the known physical law and has suitable sprite like a rocket. I wonder he didn't expect a missile firing a beam weapon.

Reply:
My vote is for "perfectly emitted." As for why it is that this only happens without acceleration, I have no idea.

Reply:
Lemme toss my hat in the pile
I'd like an attribute which can enable or disable this PR's effects (kind of. The main issue is that a weapon's submunitions, and a submunition's submunitions, act differently) so that you can choose either the unrealistic but very nice to work with PR behavior or inheriting only velocity magnitude and applying that to the submunition's direction, or the more realistic inheriting velocity magnitude and direction, and letting the submunition's own acceleration+drag to do its own moving.

I need this PR for some flashy lightning weapon, so that it could theoretically infinitely branch out (but I will only have it branch out about 6 times) and look like actually tree-like branchy lightning.